docs/usage/configuration/config_documentation.md

```diff
@@ -3765,6 +3806,109 @@ redis:
   port: 6379
   password: <secret_password>
 ```
+---
+## Worker configuration for workers configuration
+These switches configure each worker them self with the a worker configuration file.
+
+Note also the configuration for the
+[main process above](#worker-configuration-for-main-process).
+
+The manual how to configure workers is described in detail in the
+[worker documentation](../../workers.md)
+
+---
+### `worker_app`
+
+The type of worker. The currently available worker applications are listed
+in [worker documentation](../../workers.md#available-worker-applications).
+
+The most common worker is the
+[`synapse.app.generic_worker`](../../workers.md#synapseappgeneric_worker).
+
+Example configuration:
+```yaml
+worker_app: synapse.app.generic_worker
+```
+---
+### `worker_name`
+
+A unique name for the worker. The worker needs a name to be addressed in
+further parameters and identification in log files.
+
+Example configuration:
+```yaml
+worker_name: generic_worker1
+```
+---
+### `worker_replication_host`
+
+The HTTP replication endpoint that it should talk to on the main Synapse process.
+The main Synapse process defines this with a `replication` resource in
+[`listeners` option](#listeners).
+
+Example configuration:
+```yaml
+worker_replication_host: 127.0.0.1
+```
+---
+### `worker_replication_http_port`
+
+The HTTP replication port that it should talk to on the main Synapse process.
+The main Synapse process defines this with a `replication` resource in
+[`listeners` option](#listeners).
+
+Example configuration:
+```yaml
+worker_replication_http_port: 9093
+```
+---
+### `worker_listeners`
+
+A worker can handle HTTP requests. If handling HTTP requests, a `worker_listeners`
```

Comment:
I think this should read: "Some workers can handle internal HTTP requests from other workers."? Maybe even "Some privileged workers"?

Reply:
The workers can not handle internal request from other workers. They can handle external request from clients or other federations.

Reply:
> The workers can not handle internal request from other workers.

I don't think this is correct: e.g. event_creators make HTTP calls to event_persisters using `/_synapse/replication/send_event/EVENT_ID/TXN`, see [here](https://github.com/matrix-org/synapse/blob/8ab16a92edd675453c78cfd9974081e374b0f998/synapse/replication/http/send_event.py#L116-L154). In general I think `worker_listeners` needs to include a replication listener here if you're one of the workers in the `instance_map`/`stream_writers`.

But: you're absolutely right that you need to define e.g. client/federation listeners here too. I don't know if we ought to mention the replication listeners here or if that will cause further confusion.

How about:

```
A worker can handle HTTP requests. To do so, a `worker_listeners` option 
must be declared, in the same way as the [`listeners` option](#listeners) 
in the shared config.

Workers declared in [`stream_writers`](#LINK_TODO) will need to include a
`replication` listener here, in order to accept internal HTTP requests from
other workers.
```

Reply:
=> https://github.com/matrix-org/synapse/pull/14086/commits/a943440797467342ad132cadeb5d98647fc4ab53